api/app/fireweather_bot/hourly_actuals.py

```diff
@@ -63,58 +94,39 @@ def _get_end_date(self) -> int:
         """ Return now. E.g. if it's 17h15 now, we'd get YYYYMMDD17 """
         return int(self.now.strftime('%Y%m%d%H'))
 
-    def process_csv(self, filename: str):
-        with open(filename, 'r') as csv_file:
-            data_df = pd.read_csv(csv_file)
-        station_codes = get_station_names_to_codes()
-        # drop any rows where 'display_name' is not found in the station_codes lookup:
-        data_df.drop(index=data_df[~data_df['display_name'].isin(
-            station_codes.keys())].index, inplace=True)
-        # replace 'display_name' column (station name) in df with station_id
-        # and rename the column appropriately
-        data_df['display_name'].replace(station_codes, inplace=True)
-        data_df.rename(columns={'display_name': 'station_code'}, inplace=True)
-
-        # write to database using _session's engine
-        with app.db.database.get_write_session_scope() as session:
-            # write the data_df to the database one row at a time, so that if data_df contains >=1 rows that
-            # are duplicates of what is already in the db, the write won't fail for the unique rows
-            # NOTE: iterating over the data_df one Series (row) at a time is ugly, but until pandas is
-            # updated with a fix, this is the easiest work-around.
-            # See https://github.com/pandas-dev/pandas/issues/15988
-            # pylint: disable=unused-variable
-            for index, row in data_df.iterrows():
-                try:
-                    # Go from pandas to a python dict.
-                    data = row.to_dict()
-                    # Format and fix timestamp, make it be tz aware.
-                    data['weather_date'] = _fix_datetime(data['weather_date'])
-                    # Throw the data into the model, and persist in the database
-                    save_hourly_actual(session, HourlyActual(**data))
-                except IntegrityError:
-                    logger.info('Skipping duplicate record for %s @ %s',
-                                data['station_code'], data['weather_date'])
-                    session.rollback()
-
-    def construct_request_body(self):
-        return {
-            'Start_Date': self._get_start_date(),
-            'End_Date': self._get_end_date(),
-            'Format': 'CSV',
-            'cboFilters': config.get('BC_FIRE_WEATHER_FILTER_ID'),
-            'rdoReport': 'OSBH',
-        }
+    def run_wfwx(self):
```

Comment:
If you make this method async, then you can await on station and observation list - then you only need one section of code with an event loop.